constants/__init__.py

```diff
@@ -76,44 +74,65 @@
     GemmaForCausalLM,
 }
 
+M772_DATASET = "Falcon/RefinedWeb"
+B7_DATASET = "HF/FineWebEdu2"
 
-# A mapping of block numbers to ModelCriteria. Must be ordered by block.
-MODEL_CRITERIA_BY_BLOCK = [
-    (
-        0,
-        ModelCriteria(
-            sequence_length=SEQUENCE_LENGTH_1,
-            optimized=False,
-            max_model_bytes=5 * 1024 * 1024 * 1024,
-            max_model_parameters=186_000_000,
-            allowed_model_types=ALLOWED_MODEL_TYPES_1,
-            tokenizer_identifier=TokenizerIdentifier.DISTILGPT_2,
-        ),
+# Defined dataset by competition id
+DATASET_BY_COMPETITION_ID: Dict[CompetitionId, str] = {
+    CompetitionId.M772_MODEL : M772_DATASET,
+    CompetitionId.B7_MODEL : B7_DATASET, 
+}
+
+# Defined model constraints by competition id to ensure they are constant across blocks.
+MODEL_CONSTRAINTS_BY_COMPETITION_ID: Dict[CompetitionId, ModelConstraints] = {
+    CompetitionId.M772_MODEL: ModelConstraints(
+        max_model_parameter_size=772_000_000,
+        sequence_length=1024,
+        allowed_architectures=ALLOWED_MODEL_TYPES_1,
+        tokenizer="distilgpt2",
+        eval_block_delay=1200,  # ~4 hours.
     ),
-    (
-        2_405_920,
-        ModelCriteria(
-            sequence_length=SEQUENCE_LENGTH_1,
-            optimized=False,
-            max_model_bytes=5 * 1024 * 1024 * 1024,
-            max_model_parameters=772_000_000,
-            allowed_model_types=ALLOWED_MODEL_TYPES_1,
-            tokenizer_identifier=TokenizerIdentifier.DISTILGPT_2,
-        ),
+    CompetitionId.B7_MODEL: ModelConstraints(
+        max_model_parameter_size=6_900_000_000,
+        sequence_length=4096,
+        allowed_architectures=ALLOWED_MODEL_TYPES_2,
+        tokenizer="Xenova/gpt-4",
+        kwargs={
+            "torch_dtype": torch.bfloat16,
+            "attn_implementation": "flash_attention_2",            
+        },
+        eval_block_delay=1200,  # ~4 hours.
     ),
+}
+
+
+# Schedule of competitions by block.
+COMPETITION_SCHEDULE_BY_BLOCK: List[Tuple[int, List[Competition]]] = [
     (
-        BLOCK_7B,
-        ModelCriteria(
-            sequence_length=SEQUENCE_LENGTH_2,
-            optimized=True,
-            max_model_bytes=15 * 1024 * 1024 * 1024,
-            max_model_parameters=6_900_000_000,
-            allowed_model_types=ALLOWED_MODEL_TYPES_2,
-            tokenizer_identifier=TokenizerIdentifier.GPT_4_TIKTOKEN,
-        ),
-    ),
+        0,
```

Comment:
I'd recommend setting "0" for only the B7 competition and then using a future block to introduce both. This means you can decouple the release from the start of the competition to ensure better vtrust across validators when the new competition starts.